src/transformers/modeling_utils.py

```diff
@@ -6021,19 +6021,22 @@ def caching_allocator_warmup(model: PreTrainedModel, expanded_device_map: dict,
 
     # This will kick off the caching allocator to avoid having to Malloc afterwards
     for device, byte_count in total_byte_count.items():
-        if device.type == "cuda":
-            index = device.index if device.index is not None else torch.cuda.current_device()
-            device_memory = torch.cuda.mem_get_info(index)[0]
+        if device.type in ["cuda", "xpu"]:
+            torch_accelerator_module = getattr(torch, device.type, torch.cuda)
```

Comment:
Could it be safe with simply  `torch_accelerator_module = getattr(torch, device.type, torch.cuda)`.

if device.type is `xpu`, would it be even possible torch don't have `xpu` attribute?

Reply:
yes, that's why I think we don't need `torch.cuda` in 

```
torch_accelerator_module = getattr(torch, device.type, torch.cuda)
```

(and having it there seems a bit strange)

Reply:
I see your point, there is no need to add the default value.

Reply:
@yiding i see your point, you wrote "simply `torch_accelerator_module = getattr(torch, device.type, torch.cuda)`" 😅
that's why i understood the opposite